lighthouse-core/audits/autocomplete.js

```diff
@@ -43,22 +194,80 @@ class AutocompleteAudit extends Audit {
     };
   }
 
+  /**
+   * @param {LH.Artifacts.FormInput} input
+   * @return {{hasValidTokens: Boolean, inValidOrder?: Boolean}}
+   */
+  static checkAttributeValidity(input) {
```

Comment:
This still applies: https://github.com/GoogleChrome/lighthouse/pull/11342#discussion_r479461621

Reply:
so take this function out the audit class? @connorjclark

Reply:
talked it over with connor. `checkAttributeValidity` does a few things now:

1. returns `{hasValidTokens: false}` if the attribute is empty
1. checks each token
   * valid if its a section- token
   * valid if it's one of the spec tokens
1. determines if order is incorrect.

we should have unit tests to evaluate these cases.



since `checkAttributeValidity` is a standalone thing, you can construct some test cases in autocompelte-test where you create some fake LH.Artifacts.FormInput objects and then pass it in.  (we dont want end2end tests through audit.. just focusing on this specific method)

make sense?

Reply:
ohhh I see okay makes sense